iothub_client/src/iothub_client_core_ll.c

```diff
@@ -2319,6 +2340,24 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_LL_SetOption(IOTHUB_CLIENT_CORE_LL_HANDLE
                 result = IOTHUB_CLIENT_OK;
             }
         }
+        else if (strcmp(optionName, OPTION_DT_MODEL_ID) == 0)
+        {
+            if (handleData->dt_model_id != NULL)
+            {
+                STRING_delete(handleData->dt_model_id);
+                handleData->dt_model_id = NULL;
+            }
+
+            if ((handleData->dt_model_id = STRING_construct_sprintf("&%s=%s", DT_MODEL_ID_TOKEN, STRING_c_str(URL_EncodeString((const char*)value)))) == NULL)
```

Comment:
>if ((handleData->dt_model_id = STRING_construct_sprintf("&%s=%s", DT_MODEL_ID_TOKEN, STRING_c_str(URL_EncodeString((const char*)value)))) == NULL) [](start = 12, length = 146)

I get it's pedantic, but we shouldn't URL encode at this stage because if (!!) we ever wanted AMQP to callback into this, the URL encode wouldn't make sense or it.